topics/releases.md

```diff
@@ -1,144 +1,109 @@
 
-## Versioning
+Upgrading from a previous patch version should be safe and seamless.
+It should be safe to run a Valkey cluster with servers running on different patch versions.
 
-Stable releases liberally follow the usual `major.minor.patch` semantic
-versioning schema. The primary goal is to provide explicit guarantees regarding
-backward compatibility.
+*Patch* versions may also introduce small improvements such as performance or memory optimizations that are easy to verify as safe.
 
-### Patch-Level versions
+### Minor versions
 
-Patches primarily consist of bug fixes and very rarely introduce any
-compatibility issues.
+*Minor* versions are released with new functionality that is added in a backward compatible manner.
+Examples of new functionality include new commands, info fields, or configuration parameters.
 
-Upgrading from a previous patch-level version is almost always safe and
-seamless.
+Upgrading from a previous minor version should be safe, and will not introduce incompatibilities between servers in the cluster.
 
-New features and configuration directives may be added, or default values
-changed, as long as these don’t carry significant impacts or introduce
-operations-related issues.
+**NOTE:** Minor releases may include new commands and data types that can introduce incompatibility between servers in the cluster, but users need to opt-in to these features to cause this type of incompatibility.
+For this reason, it is not recommended to run a Valkey cluster with servers running on different minor versions.
+Users should avoid new features until all servers in the cluster have been upgraded.
 
-### Minor versions
+Commands may also be deprecated in minor versions.
+If a command is deprecated, a replacement command or an alternative to using the command will be defined in the same minor version.
 
-Minor versions usually deliver maturity and extended functionality.
+### Major versions
 
-Upgrading between minor versions does not introduce any application-level
-compatibility issues.
+*Major* versions are released with significant functionality that may break backwards compatibility or alter key performance characteristics.
+Examples of significant functionality includes altering the behavior of an existing command, removing previously deprecated commands, changing the default value of configs, and significant refactoring for performance improvements.
 
-Minor releases may include new commands and data types that introduce
-operations-related incompatibilities, including changes in data persistence
-format and replication protocol.
+Upgrading from a previous major version is intended to be safe, but should be approached with caution. 
+You should carefully read the release notes before performing a major version upgrade.
+Major version upgrades do not guarantee backwards compatibility, which means you should always upgrade replicas before upgrading primaries in order to ensure data consistency.
```

Comment:
>Major version upgrades do not guarantee backwards compatibility, which means you should always upgrade replicas before upgrading primaries in order to ensure data consistency.

If we don't guarantee "backwards compatibility", how do we guarantee users can always replicate a older version of Valkey to the latest major version? I wonder if we should take the similar approach like in the minor version upgrade case where potentially incompatible changes are gated behind a server config?

Reply:
I needed to rework this section, since we have some odd guarantees here.